snap/frontend/components/Navbar/index.jsx

```diff
@@ -29,13 +29,21 @@ const Navbar = () => {
 
 		setSelectedNetwork(option.label);
 
-		dispatch({ type: actionTypes.SET_NETWORK, payload: networkData });
+		dispatch.setNetwork(networkData);
 	};
 
 	const handleSelectCurrency = option => {
 		setSelectedCurrency(option.label);
 	};
 
+	useEffect(() => {
+		// set selected network in dropdown label
+		if (network && network.hasOwnProperty('networkName')) {
+			const selected = networks.find(n => n.value === network.networkName);
```

Comment:
What happens if the network is not found?

Reply:
I added logic to handle it.
It will skip to update network name instead of hit error.